### PR TITLE
[SPARK-29528][BUILD] Upgrade scala-maven-plugin to 4.3.0 for Scala 2.13.1

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to upgrade `scala-maven-plugin` to `4.3.0` for Scala `2.13.1`.
We tried 4.2.4, but it's reverted due to Windows build issue. Now, `4.3.0` has a Window fix.

### Why are the changes needed?
Scala 2.13.1 seems to break the binary compatibility.

We need to upgrade `scala-maven-plugin` to bring the the following fixes for the latest Scala 2.13.1.
- https://github.com/davidB/scala-maven-plugin/issues/363
- https://github.com/sbt/zinc/issues/698

Also, `4.3.0` has the following Window fix.
- https://github.com/davidB/scala-maven-plugin/issues/370 (4.2.4 throws error on Windows)

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

- For now, we don't support Scala-2.13. This PR at least needs to pass the existing Jenkins with Maven to get prepared for Scala-2.13.
- `AppVeyor` passed. (https://ci.appveyor.com/project/ApacheSoftwareFoundation/spark/builds/28745383)

### CODE REVIEW

File: R/README.md
Comment (dongjoon-hyun) [inline R/README.md]: This is only for triggering AppVoyer and this will be reverted before merging.

**Comment by HyukjinKwon** [comment]
Thanks @dongjoon-hyun for trying this out!

**Comment by dongjoon-hyun** [comment]
Thank you. AppVeyor passed!
Please ignore github action failure.
It will be fixed when my another PR about GitHub Action Cache PR is merged.

**Review by HyukjinKwon — APPROVED** [review]
LGTM if tests pass

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for review and approval. Merged to master.
This passes Jenkins and AppVeyor.

**Pull request closed without merging**
